### DISCUSSION TITLE
Sample scaledjob template for Selenium Grid Scaler

### DISCUSSION THREAD

**Discussion by abhitrivedi2006** [kedacore/keda#3910] [Q&A / Need Help]
Hello There,

I want to create scaledjob for selenium grid scaler .. can you pls share any sample template to configure that.

**Comment by tomkerkhove** [comment]
You can find some on https://keda.sh/docs/latest/scalers/selenium-grid-scaler/

**Comment by abhitrivedi2006** [comment]
yes I saw that .. I am trying to understand how scaledjob works with Selenium grid scalar. this is what I have configured.


apiVersion: keda.sh/v1alpha1
kind: ScaledJob
metadata:
  name: selenium-grid-chrome-scaledjob
spec:
  jobTargetRef:
    parallelism: 1                            
    completions: 1                           
    activeDeadlineSeconds: 600                
    backoffLimit: 6                           
    template:
     spec:
      volumes:
        - name: dshm
          emptyDir:
            medium: Memory
            sizeLimit: 2Gi
      containers:
        - name: selenium-chrome-node
          image: selenium/node-chrome:4.6.0-20221104
          ports:
            - containerPort: 5555
              protocol: TCP
          envFrom:
            - configMapRef:
                name: selenium-event-bus-config
          resources:
            limits:
              cpu: '1'
              memory: 1Gi
            requests:
              cpu: '1'
              memory: 1Gi
          volumeMounts:
            - name: dshm
              mountPath: /dev/shm
          startupProbe:
            httpGet:
              path: /status
              port: 5555
              scheme: HTTP
            timeoutSeconds: 1
            periodSeconds: 5
            successThreshold: 1
            failureThreshold: 120
          lifecycle:
            preStop:
              exec:
                command:
                  - bash
                  - '-c'
                  - >
                    curl -X POST 127.0.0.1:5555/se/grid/node/drain --header
                    'X-REGISTRATION-SECRET;' && \

                    while curl 127.0.0.1:5555/status; do sleep 1; done
      terminationGracePeriodSeconds: 3600    
  pollingInterval: 30                         
  successfulJobsHistoryLimit: 5               
  failedJobsHistoryLimit: 5                   
#  envSourceContainerName: {container-name}    
  minReplicaCount: 1                        
  maxReplicaCount: 75                        
  rolloutStrategy: gradual                    
  rollout:
    strategy: gradual                         
    propagationPolicy: foreground             
  scalingStrategy:
    strategy: "custom"                        
    customScalingQueueLengthDeduction: 1      
    customScalingRunningJobPercentage: "0.5"  
    pendingPodConditions:                     
      - "Ready"
      - "PodScheduled"
      - "AnyOtherCustomPodCondition"
    multipleScalersCalculation : "max" 
  triggers:
   - type: selenium-grid
     metadata:
        url: 'http://selenium-router.connected.svc.cluster.local:4444/graphql'
        browserName: 'chrome'
______________________________________________
Though I have configured maxReplicaCount: 75  but i see container counts goes beyond 100.
![image](https://user-images.githubusercontent.com/60986284/204021970-0753de8f-120e-477c-8a25-31966bf5b2cb.png)

**Comment by Fuut2000** [comment]
Hello, this can be solved by reducing the maxReplicaCount to 65% of the intended maxReplicaCount. Perhaps because of Kubernetes termination period? I also found out that in the latest grid 4 versions the number of containers in the lower left in de ui are higher than the actual shown containers. Do you have multiple scalers, 1 for each browser?  Acoording to the setting multipleScalersCalculation , Keda cannot scale multiple scalers to the exact number of corresponding requests for each stereotype. You always have to choose between 1 of the scalers metrics of use an avarage or sum. It's unclear to me what is specifically meant by "metrics".
